core/src/main/scala/org/apache/spark/deploy/worker/Worker.scala

```diff
@@ -538,10 +539,10 @@ private[spark] object Worker extends Logging {
       memory: Int,
       masterUrls: Array[String],
       workDir: String,
-      workerNumber: Option[Int] = None): (ActorSystem, Int) = {
+      workerNumber: Option[Int] = None,
+      conf: SparkConf = new SparkConf): (ActorSystem, Int) = {
```

Comment:
This constructor with a custom `conf` is only called from LocalSparkCluster.   We need to use the supplied SparkConf here so that the environment variable mocking works correctly.